railties/test/generators/db_system_change_generator_test.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require "generators/generators_test_helper"
+require "rails/generators/rails/db/system/change/change_generator"
+
+module Rails
+  module Generators
+    module Db
+      module System
+        class ChangeGeneratorTest < Rails::Generators::TestCase
+          include GeneratorsTestHelper
+
+          setup :copy_gemfile
+
+          test "change to invalid database" do
+            output = capture(:stderr) do
+              run_generator ["--to", "invalid-db"]
+            end
+            assert_match <<~MSG.squish, output
+              Invalid value for --to option.
+              Supported preconfigurations are:
+              mysql, postgresql, sqlite3, oracle, frontbase,
+              ibm_db, sqlserver, jdbcmysql, jdbcsqlite3,
+              jdbcpostgresql, jdbc.
+            MSG
+          end
+
+          test "change to postgresql" do
+            output = run_generator ["--to", "postgresql"]
+            assert_file("config/database.yml") do |content|
+              assert_match "adapter: postgresql", content
+              assert_match "database: test_app", content
+            end
+            assert_file("Gemfile") do |content|
```

Comment:
Adding a newline after `output` and between the `assert_file` seems like it would increase readability in these tests.